docs/CONTRIBUTING.md

```diff
@@ -46,13 +46,60 @@ make update-assets
 
 Commit and submit a PR.
 
-## Commit guidelines
-
-The title of the commit message should describe the _what_ about the
-changes. Additionally, it is helpful to add the _why_ in the body of
-the commit changes. Make sure to include any assumptions that you
-might have made in this commit. Changes to unrelated parts of the
-codebase should be kept as separate commits.
+## Authoring PRs
+
+These are general guidelines for making PRs/commits easier to review. Use your own
+judgement to decide when it is best to follow them. These are not strict rules
+but highly recommended to follow:
+
+ * Commits should be atomic and self-contained. Divide logically separate changes
+   to separate commits. This principle  is best explained in the the Linux Kernel
```

Comment:
Any specific reason for double space here?
```suggestion
   to separate commits. This principle is best explained in the the Linux Kernel
```

Reply:
Nope :see_no_evil: